authorization-common/src/main/java/com/sequenceiq/authorization/utils/GetAuthzActionTypeProvider.java

```diff
@@ -0,0 +1,89 @@
+import static com.sequenceiq.authorization.resource.AuthorizationResourceAction.DESCRIBE_DATALAKE;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceAction.DESCRIBE_ENVIRONMENT;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceAction.DESCRIBE_IMAGE_CATALOG;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceAction.DESCRIBE_PROXY;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceAction.DESCRIBE_RECIPE;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceAction.STRUCTURED_EVENTS_READ;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.AUDIT_CREDENTIAL;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.CLUSTER_DEFINITION;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.CLUSTER_TEMPLATE;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.CREDENTIAL;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.CUSTOM_CONFIGURATIONS;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.DATABASE;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.DATABASE_SERVER;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.DATAHUB;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.DATALAKE;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.ENVIRONMENT;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.FREEIPA;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.IMAGE_CATALOG;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.KERBEROS;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.LDAP;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.PROXY;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.RECIPE;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.STRUCTURED_EVENT;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sequenceiq.authorization.resource.AuthorizationResourceAction;
+import com.sequenceiq.authorization.resource.AuthorizationResourceType;
+
+public final class GetAuthzActionTypeProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GetAuthzActionTypeProvider.class);
+
+    private static final Map<AuthorizationResourceType, Set<AuthorizationResourceAction>> PAIRS = new HashMap<>(AuthorizationResourceType.values().length);
+
+    static {
+        PAIRS.put(CREDENTIAL, Set.of(DESCRIBE_CREDENTIAL, DESCRIBE_CREDENTIAL_ON_ENVIRONMENT));
```

Comment:
DESCRIBE_CREDENTIAL and DESCRIBE_CREDENTIAL_ON_ENVIRONMENT has the same permission value, the only difference is that we use the second one on environment APIs if the crn belongs to environment. in this case DESCRIBE_CREDENTIAL should be enough, in general there is no need to use set, only 1 action/resource should be enought here